src/TaskEither.ts

```diff
@@ -272,8 +270,16 @@ export const getOrElseW: <E, B>(
  * @category interop
  * @since 2.0.0
  */
-export const tryCatch = <E, A>(f: Lazy<Promise<A>>, onRejected: (reason: unknown) => E): TaskEither<E, A> => () =>
-  f().then(_.right, (reason) => _.left(onRejected(reason)))
+export const tryCatch = <E, A>(
+  f: Lazy<Promise<A>>,
+  onRejected: (reason: unknown) => E
+): TaskEither<E, A> => async () => {
+  try {
+    return await f().then(_.right)
```

Comment:
Isn't return await an anti-pattern and causes performance issues (https://arthur.place/the-cost-of-return-await)?

Reply:
return-await is necessary in this case to catch promise rejections (there's a section about this in the article you cited)

Reply:
Oh, good catch!